### ISSUE TITLE
Compile Rocket v0.4.10 failed due to errors in devise_core v0.2.0 with rustc v1.63.0-nightly (2022-05-24)

### ISSUE THREAD

**Issue by Ronaldho80** [SergioBenitez/Rocket#2205]
**Description**

After upgrade of rustc to version v1.63.0-nightly 2022-05-24 I get compile errors for devise_core v0.2.0:

```
error[E0557]: feature has been removed
 --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/lib.rs:2:12
  |
2 | #![feature(crate_visibility_modifier)]
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^ feature has been removed
  |
  = note: removed in favor of `pub(crate)`

error: expected one of `!` or `::`, found keyword `enum`
  --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/field.rs:52:7
   |
52 | crate enum FieldKind<'p> {
   |       ^^^^ expected one of `!` or `::`

error: expected identifier, found keyword `crate`
  --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:69:5
   |
69 |     crate generic_support: GenericSupport,
   |     ^^^^^ expected identifier, found keyword

error: expected one of `!` or `::`, found keyword `fn`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:28:15
    |
28  |           crate fn push_default_mappers(&mut self) {
    |                 ^^ expected one of `!` or `::`
...
193 | /     mappers! {
194 | |         (map_struct, try_map_struct, struct_mapper): Struct, struct_mappers,
195 | |         (map_enum, try_map_enum, enum_mapper): Enum, enum_mappers,
196 | |         (map_variant, try_map_variant, variant_mapper): Variant, variant_mappers,
197 | |         (map_fields, try_map_fields, fields_mapper): Fields, fields_mappers,
198 | |         (map_field, try_map_field, field_mapper): Field, field_mappers
199 | |     }
    | |_____- in this macro invocation
    |
    = note: this error originates in the macro `mappers` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0432]: unresolved imports `field::Field`, `field::Fields`
 --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:8:13
  |
8 | use field::{Field, Fields};
  |             ^^^^^  ^^^^^^ no `Fields` in `field`
  |             |
  |             no `Field` in `field`

error[E0432]: unresolved imports `field::Field`, `field::FieldParent`, `field::Fields`
 --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/derived.rs:5:13
  |
5 | use field::{Field, FieldParent, Fields};
  |             ^^^^^  ^^^^^^^^^^^  ^^^^^^ no `Fields` in `field`
  |             |      |
  |             |      no `FieldParent` in `field`
  |             no `Field` in `field`

   Compiling ghash v0.3.1
error[E0599]: no method named `variant_mapper` found for reference `&DeriveGenerator` in the current scope
  --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:90:22
   |
90 |         .map(|v| gen.variant_mapper()(gen, v))
   |                      ^^^^^^^^^^^^^^ method not found in `&DeriveGenerator`

error[E0599]: no method named `variant_mapper` found for reference `&DeriveGenerator` in the current scope
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:106:22
    |
106 |         .map(|v| gen.variant_mapper()(gen, v))
    |                      ^^^^^^^^^^^^^^ method not found in `&DeriveGenerator`

error[E0599]: no method named `fields_mapper` found for reference `&DeriveGenerator` in the current scope
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:113:9
    |
113 |     gen.fields_mapper()(gen, data.fields())
    |         ^^^^^^^^^^^^^ method not found in `&DeriveGenerator`

   Compiling sha2 v0.10.2
error[E0599]: no method named `fields_mapper` found for reference `&DeriveGenerator` in the current scope
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:117:9
    |
117 |     gen.fields_mapper()(gen, data.fields())
    |         ^^^^^^^^^^^^^ method not found in `&DeriveGenerator`

error[E0599]: no method named `field_mapper` found for reference `&DeriveGenerator` in the current scope
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:126:24
    |
126 |         .map(|field| g.field_mapper()(g, field))
    |                        ^^^^^^^^^^^^ method not found in `&DeriveGenerator`

error[E0615]: attempted to take value of method `generic_support` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:159:14
    |
159 |         self.generic_support = support;
    |              ^^^^^^^^^^^^^^^ method, not a field
    |
    = help: methods are immutable and cannot be assigned to

error[E0615]: attempted to take value of method `data_support` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:164:14
    |
164 |         self.data_support = support;
    |              ^^^^^^^^^^^^ method, not a field
    |
    = help: methods are immutable and cannot be assigned to

error[E0609]: no field `type_generic_mapper` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:171:14
    |
171 |         self.type_generic_mapper = Some(Box::new(f));
    |              ^^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `generic_replacements` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:176:14
    |
176 |         self.generic_replacements.push((trait_gen, impl_gen));
    |              ^^^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `functions` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:188:14
    |
188 |         self.functions.push(Box::new(f));
    |              ^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0599]: no method named `push_default_mappers` found for mutable reference `&mut DeriveGenerator` in the current scope
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:189:14
    |
189 |         self.push_default_mappers();
    |              ^^^^^^^^^^^^^^^^^^^^ method not found in `&mut DeriveGenerator`

error[E0615]: attempted to take value of method `data_support` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:206:56
    |
206 |         let (span, support) = (self.input.span(), self.data_support);
    |                                                        ^^^^^^^^^^^^ method, not a field
    |
help: use parentheses to call the method
    |
206 |         let (span, support) = (self.input.span(), self.data_support(_));
    |                                                                    +++

error[E0615]: attempted to take value of method `generic_support` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:231:57
    |
231 |             let (span, support) = (generic.span(), self.generic_support);
    |                                                         ^^^^^^^^^^^^^^^ method, not a field
    |
help: use parentheses to call the method
    |
231 |             let (span, support) = (generic.span(), self.generic_support(_));
    |                                                                        +++

error[E0609]: no field `generics_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:247:15
    |
247 |         (self.generics_validator)(self, &self.input.generics)?;
    |               ^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `struct_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:251:23
    |
251 |                 (self.struct_validator)(self, derived)?;
    |                       ^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `fields_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:252:23
    |
252 |                 (self.fields_validator)(self, derived.fields())?;
    |                       ^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `enum_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:256:23
    |
256 |                 (self.enum_validator)(self, derived)?;
    |                       ^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `fields_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:258:27
    |
258 |                     (self.fields_validator)(self, variant.fields())?;
    |                           ^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `functions` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:267:26
    |
267 |         for i in 0..self.functions.len() {
    |                          ^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `functions` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:268:34
    |
268 |             let function = &self.functions[i];
    |                                  ^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `struct_mappers` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:272:26
    |
272 |                     self.struct_mappers[i](self, derived)?
    |                          ^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `enum_mappers` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:276:26
    |
276 |                     self.enum_mappers[i](self, derived)?
    |                          ^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `type_generic_mapper` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:288:45
    |
288 |         if let Some(ref type_mapper) = self.type_generic_mapper {
    |                                             ^^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `generic_replacements` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:302:36
    |
302 |             let replacement = self.generic_replacements.iter()
    |                                    ^^^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `enum_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:180:37
    |
180 |     validator!(validate_enum: Enum, enum_validator);
    |                                     ^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `struct_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:181:41
    |
181 |     validator!(validate_struct: Struct, struct_validator);
    |                                         ^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `generics_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:182:51
    |
182 |     validator!(validate_generics: &syn::Generics, generics_validator);
    |                                                   ^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`

error[E0609]: no field `fields_validator` on type `&mut DeriveGenerator`
   --> /home/herzog/.cargo/registry/src/github.com-1ecc6299db9ec823/devise_core-0.2.0/src/generator.rs:183:41
    |
183 |     validator!(validate_fields: Fields, fields_validator);
    |                                         ^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `input`, `trait_impl`, `trait_path`


```

**To Reproduce**
Cargo.toml:
```
[package]
name = "ltx_server"
edition = "2021"
version = "0.1.0"
authors = ["Dr. Ronny Herzog <herzog@lipotype.com>"]

[dependencies]
#rocket_cors = "0.4"
chrono = { version = "0.4", features = ["serde"] }
color-backtrace = { version = "0.1" }
config = "0.8"
crossbeam-channel = "*"
diesel = { version = "1.4", features = ["sqlite", "chrono", "r2d2"] }
failure = "*"
failure_derive = "*"
flexi_logger = "0.11"
frank_jwt = "3.0.0"
lazy_static = "*"
log = "0.4"
multipart = "=0.16.1"
num_cpus = "*"
regex = "1"
rocket = "0.4"
rocket_cors = "0.5"
serde = "1.0"
serde_derive = "1.0"
serde_json = "1.0"
sha2 = "*"
slugify = "0.1.0"
threadpool = "*"
time = "*"
quote = "=1.0.3"
ws = "*"
wait-timeout = "*"
rocket_upload = "*"
zip = "0.5"
toml = "*"
git2 = "*"
walkdir = "2"

# Lipotype dependencies
mfql = { path = "../backend/mfql" }
identifier = { path = "../backend/identifier" }
masterscan = { path = "../backend/masterscan" }
xplorer_config = { path = "../backend/config" }

[dependencies.rocket_contrib]
version = "0.4"
default-features = false
features = ["json", "diesel_sqlite_pool"]

[dev-dependencies]
time = "*"
docopt = "1.0"
speculate = "*"
reqwest = "*"

[build-dependencies]
vergen = "5"
```

**Expected Behavior**

It should compile

**Environment:**

- Ubuntu 20.04.4
- Rocket v0.4
- Rust v1.63.0

**Comment by bjauny** [comment]
The issue is also present when trying to build v0.5.0.rc.2, which still tries to use devise_core v0.2.0.

**Comment by SergioBenitez** [comment]
> The issue is also present when trying to build v0.5.0.rc.2, which still tries to use devise_core v0.2.0.

That can't be. Rocket 0.5 builds with Rust stable, but this issue is caused by Rocket/Pear/Devise making use of a nightly feature. So if what you're saying were true, Rocket 0.5 would not build on stable.

**Comment by SergioBenitez** [comment]
Fixed in 5824f56c4efa50dc989a995fda7a73229f99d744. Pushing out the 0.4.11 release soon.

**Comment by SergioBenitez** [comment]
0.4.11 is published.
